internal/putil/putil.go

```diff
@@ -264,3 +289,42 @@ func Walk(v property.Value, f func(property.Value) (continueWalking bool)) bool
 	}
 	return true
 }
+
+// GetPropertyDependencies gathers (deeply) the dependencies of the given property value.
+func GetPropertyDependencies(v property.Value) []urn.URN {
+	var deps []urn.URN
+	Walk(v, func(v property.Value) (continueWalking bool) {
+		deps = append(deps, v.Dependencies()...)
+		return true
+	})
+	slices.Sort(deps)
+	return slices.Compact(deps)
+}
+
+// MergePropertyDependencies merges the given dependencies into the property map.
+//
+// Apply the dependencies to the property map, only for "legacy" values that don't have output dependencies.
+// The caller may fold the output dependencies of a value's children into the dependencies map,
+// and we seek to avoid treating those as true dependencies of the top-level value.
```

Comment:
I don't know what this means.

Are you saying, given a map `m` and a dependency map `deps`: 
```
m = { Dependencies: ["urn1"], Values: { "n": { Dependencies: ["urn2"] } } }
deps = { "m": ["urn3"] }
```

that we want to see 

```
MergePropertyDependencies(m, deps) = 
	{ Dependencies: ["urn1", "urn3"], Values: { "n": { Dependencies: ["urn2"] } } }
```

An example here would make it much easier to understand.

Reply:
Yes. See the test cases in putil_test.go for some examples.